apps/linkedin-to-notion/src/contents/linkedin-notion-side-panel.tsx

```diff
@@ -18,6 +25,45 @@ export const getStyle: PlasmoGetStyle = () => {
 
 const LinkedinNotionSidePanel = () => {
   const [isOpen, setIsOpen] = useState(true); // set-back to false before deployment
+  const [user, setUser] = useStorage<User>({
+    key: 'user',
+    instance: new Storage({
+      area: 'local',
+    }),
+  });
+
+  useEffect(() => {
+    async function init() {
+      const { data, error } = await supabase.auth.getSession();
+
+      if (error) {
+        console.error(error);
+        return;
+      }
+      if (data.session) {
+        setUser(data.session.user);
+        sendToBackground({
+          name: 'users/resolvers/init-session',
+          body: {
+            refresh_token: data.session.refresh_token,
+            access_token: data.session.access_token,
+          },
+        });
+      }
+    }
+
+    init();
+  }, []);
+
+  const handleOAuthLogin = async (provider: Provider, scopes = 'email') => {
+    await supabase.auth.signInWithOAuth({
+      provider,
+      options: {
+        redirectTo: window.location.href,
```

Comment:
Let's assume that we want to sign up on a page with the following url: "https://www.linkedin.com/in/bastien-némésis/#".

Then, the sign up fails:
![image](https://github.com/Bastien-and-Gauvain/monorepo/assets/45038783/45037e43-7108-412a-8509-e06e7adb3a80)

I think it's due to the incongruence between that url (and the resulting redirect url) and those in the authentication parameters of supabase:
![image](https://github.com/Bastien-and-Gauvain/monorepo/assets/45038783/3f622cbd-1c2e-4683-9398-ef5daef723cc)

That's why I suggest the following changes to make sure there are no parasitic trailing symbols:
```suggestion
      redirectTo: window.location.href.match(/https:\/\/[a-z]{2,3}\.linkedin\.com\/in\/[^/]+\//)[0],
```

If we follow up on the example above, the regex pattern ensures that **we remove the trailing "#" from the url** for instance.

Reply:
Good catch!